### PR TITLE
Declare missing dependencies for FSE

### PR DESCRIPTION
**Pull request by scinos**
### Problem

`@yarnpkg/doctor` found a bunch of undeclared dependencies and missing peer dependencies:

```
➤ YN0000: ┌ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/package.json
➤ YN0000: │ User scripts prefixed with "pre" or "post" (like "posts-list-block") will not be called in sequence anymore; prefer calling prologues and epilogues explicitly
➤ YN0000: │ User scripts prefixed with "pre" or "post" (like "premium-content") will not be called in sequence anymore; prefer calling prologues and epilogues explicitly
➤ YN0000: │ User scripts prefixed with "pre" or "post" (like "prebuild") will not be called in sequence anymore; prefer calling prologues and epilogues explicitly
➤ YN0000: │ User scripts prefixed with "pre" or "post" (like "predev") will not be called in sequence anymore; prefer calling prologues and epilogues explicitly
➤ YN0000: │ User scripts prefixed with "pre" or "post" (like "prepare") will not be called in sequence anymore; prefer calling prologues and epilogues explicitly
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/bin/js-unit-config.js:18:18: Undeclared dependency on @wordpress/scripts
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/bin/js-unit-setup.js:5:1: Undeclared dependency on @testing-library/jest-dom
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/bin/npm-run-build.js:8:16: Undeclared dependency on npm-run-all
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/e2e-test-helpers/activate-theme.js:5:28: Undeclared dependency on @wordpress/e2e-test-utils
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/editor-domain-picker/index.ts:5:1: Undeclared dependency on a8c-fse-common-data-stores
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/index.ts:7:1: Undeclared dependency on a8c-fse-common-data-stores
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/editor-site-launch/index.ts:7:1: Undeclared dependency on a8c-fse-common-data-stores
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/dotcom-fse/tests/fse-back-button.spec.js:5:27: Undeclared dependency on @wordpress/e2e-test-utils
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js:18:1: Undeclared dependency on @wordpress/nux
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nux/src/disable-core-nux.js:6:1: Undeclared dependency on @wordpress/nux
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/dotcom-fse/blocks/navigation-menu/edit.js:6:1: Undeclared dependency on @wordpress/server-side-render
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/editor-domain-picker/src/stores/index.ts:4:1: Undeclared dependency on a8c-fse-common-data-stores
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/src/stores/index.ts:4:1: Undeclared dependency on a8c-fse-common-data-stores
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/editor-site-launch/src/stores/index.ts:4:1: Undeclared dependency on a8c-fse-common-data-stores
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/newspack-blocks/synced-newspack-blocks/blocks/carousel/create-swiper.js:4:1: Undeclared dependency on @wordpress/a11y
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/newspack-blocks/synced-newspack-blocks/blocks/carousel/create-swiper.js:5:1: Undeclared dependency on @wordpress/escape-html
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/newspack-blocks/synced-newspack-blocks/blocks/carousel/create-swiper.js:7:1: Undeclared dependency on swiper
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/newspack-blocks/synced-newspack-blocks/blocks/carousel/create-swiper.js:8:1: Undeclared dependency on swiper
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/test/template-selector-control.test.js:6:1: Undeclared dependency on @testing-library/react
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/test/template-selector-preview.test.js:6:1: Undeclared dependency on @testing-library/react
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/package.json:96:32: Unmet transitive peer dependency on enzyme@^3.11.0, via @automattic/calypso-build@*
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/package.json:138:31: Unmet transitive peer dependency on @babel/core@^7.0.0-0, via @babel/preset-typescript@^7.8.3
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/package.json:139:31: Unmet transitive peer dependency on eslint@6.x, via @wordpress/eslint-plugin@^6.0.0
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso2/apps/full-site-editing/package.json:140:17: Unmet transitive peer dependency on @babel/core@^7.0.0, via babel-jest@^26.0.1
➤ YN0000: └ Completed in 21.88s
```

### Changes

This PR adds those dependencies to `./apps/full-site-editing/package.json`. The version ranges have been copied from `./package.json` to make sure nothing changes.

### Testing instructions

Verify files have not actually changed:
* Checkout this branch in directory `wp-calypso` and master in `wp-calypso2`
* Run `yarn` in both directories
* Run `cd apps/full-site-editing && yarn build` in both directories
* Compare the generated files and verify they are the same in both directories: `for file in wp-calypso/apps/full-site-editing/full-site-editing-plugin/*/dist/*; do diff $file ${file/wp-calypso/wp-calypso2}; done`. The only changes should be those related with the commit hash.

Verify there are no more missing packages:
* Run `npx @yarnpkg/doctor` inside `./apps/full-site-editing`. The result should be similar to:

```
➤ YN0000: ┌ /Users/sergio/src/automattic/wp-calypso/apps/full-site-editing/package.json
➤ YN0000: │ User scripts prefixed with "pre" or "post" (like "posts-list-block") will not be called in sequence anymore; prefer calling prologues and epilogues explicitly
➤ YN0000: │ User scripts prefixed with "pre" or "post" (like "premium-content") will not be called in sequence anymore; prefer calling prologues and epilogues explicitly
➤ YN0000: │ User scripts prefixed with "pre" or "post" (like "prebuild") will not be called in sequence anymore; prefer calling prologues and epilogues explicitly
➤ YN0000: │ User scripts prefixed with "pre" or "post" (like "predev") will not be called in sequence anymore; prefer calling prologues and epilogues explicitly
➤ YN0000: │ User scripts prefixed with "pre" or "post" (like "prepare") will not be called in sequence anymore; prefer calling prologues and epilogues explicitly
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso/apps/full-site-editing/full-site-editing-plugin/editor-domain-picker/index.ts:5:1: Undeclared dependency on a8c-fse-common-data-stores
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso/apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/index.ts:7:1: Undeclared dependency on a8c-fse-common-data-stores
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso/apps/full-site-editing/full-site-editing-plugin/editor-site-launch/index.ts:7:1: Undeclared dependency on a8c-fse-common-data-stores
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso/apps/full-site-editing/full-site-editing-plugin/editor-domain-picker/src/stores/index.ts:4:1: Undeclared dependency on a8c-fse-common-data-stores
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso/apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/src/stores/index.ts:4:1: Undeclared dependency on a8c-fse-common-data-stores
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso/apps/full-site-editing/full-site-editing-plugin/editor-site-launch/src/stores/index.ts:4:1: Undeclared dependency on a8c-fse-common-data-stores
➤ YN0000: └ Completed in 13.95s
```

### CODE REVIEW

**Comment by scinos** [comment]
@yansern @noahtallen @alshakero I added you as reviewers based on GitHub suggestions. Please let me know if you think someone else should review this change.

File: apps/full-site-editing/.eslintrc.js [hunk 1]
```diff
@@ -1,8 +1,6 @@
 module.exports = {
-	plugins: [ 'jest' ],
```
Comment (scinos) [inline apps/full-site-editing/.eslintrc.js:2]: Already declared in parent `.eslintrc`. Having it here makes eslint unhappy

File: apps/full-site-editing/.eslintrc.js [hunk 2]
```diff
@@ -1,8 +1,6 @@
...
-	extends: [ 'plugin:jest/recommended', 'plugin:@wordpress/eslint-plugin/i18n' ],
+	extends: [ 'plugin:@wordpress/eslint-plugin/i18n' ],
 	rules: {
-		'import/no-extraneous-dependencies': [ 'error', { packageDir: __dirname } ],
```
Comment (scinos) [inline apps/full-site-editing/.eslintrc.js:5]: No need for this exception anymore.

File: apps/full-site-editing/full-site-editing-plugin/premium-content/blocks/container/tab.js [hunk 3]
```diff
@@ -1,20 +1,14 @@
 /**
- *
```
Comment (scinos) [inline apps/full-site-editing/full-site-editing-plugin/premium-content/blocks/container/tab.js:2]: Not sure where this changes comes from... when I run `yarn`, the file got changed automatically 🤷

File: apps/full-site-editing/full-site-editing-plugin/premium-content/blocks/logged-out-view/index.js [hunk 4]
```diff
@@ -18,9 +18,9 @@ const name = 'premium-content/logged-out-view';
 const category = getCategoryWithFallbacks( 'design', 'common' );
 /**
  * @typedef {object} Attributes
- *
```
Comment (scinos) [inline apps/full-site-editing/full-site-editing-plugin/premium-content/blocks/logged-out-view/index.js:21]: Not sure where this changes comes from... when I run `yarn`, the file got changed automatically 🤷

File: apps/full-site-editing/full-site-editing-plugin/premium-content/blocks/container/tab.js (hunk 3 above)
Comment (noahtallen) [reply apps/full-site-editing/full-site-editing-plugin/premium-content/blocks/container/tab.js:2]: Maybe an updated lint rule dealing with jsdoc?

**Comment by scinos** [comment]
> Do we need to be moving away from pre* scripts with yarn??

In a future, yes. The tool I'm using was created by yarn team to ease the transition to Yarn 2.0, which won't support pre/post scripts in some cases (https://yarnpkg.com/advanced/lifecycle-scripts)

**Pull request merged**